kyo-core/shared/src/main/scala/kyo/tags.scala

```diff
@@ -1,41 +1,206 @@
 package kyo
 
-import izumi.reflect.dottyreflection.TypeInspections
-import izumi.reflect.macrortti.LightTypeTag
+import scala.annotation.tailrec
 import scala.quoted.*
 
-opaque type Tag[T] = String
+case class Tag[T](tpe: String) extends AnyVal
 
 object Tag:
 
-    given canEqual[T, U]: CanEqual[Tag[T], Tag[U]] = CanEqual.derived
+    import internal.*
+
+    inline given apply[T >: Nothing]: Tag[T] = ${ tagImpl[T] }
+
+    extension [T](t1: Tag[T])
+
+        def show = t1.tpe.takeWhile(_ != ';')
+
+        def <:<[U](t2: Tag[U]): Boolean =
+            t1.tpe == t2.tpe || isSubtype(t1, t2)
+
+        def =:=[U](t2: Tag[U]): Boolean =
+            t1.tpe == t2.tpe
+
+        def =!=[U](t2: Tag[U]): Boolean =
+            t1.tpe != t2.tpe
+
+        def >:>[U](t2: Tag[U]): Boolean =
+            t1.tpe == t2.tpe || isSubtype(t2, t1)
 
-    extension [T](t: Tag[T])
-        def parse: LightTypeTag =
-            val arr = t.split('|')
-            LightTypeTag.parse(arr(0).toInt, arr(1), arr(2), arr(3).toInt)
     end extension
 
-    inline given apply[T]: Tag[T] = ${ tagImpl[T] }
-
-    private def tagImpl[T: Type](using Quotes): Expr[Tag[T]] =
-        import quotes.reflect.*
-        if TypeRepr.of[T] =:= TypeRepr.of[Nothing] then
-            report.errorAndAbort("Can't infer a Tag[Nothing]")
-        if TypeRepr.of[T] =:= TypeRepr.of[Any] then
-            report.errorAndAbort("Can't infer a Tag[Any]")
-        if TypeRepr.of[T] =:= TypeRepr.of[Object] then
-            report.errorAndAbort("Can't infer a Tag[Object]")
-        val ref         = TypeInspections.apply[T]
-        val fullDb      = TypeInspections.fullDb[T]
-        val nameDb      = TypeInspections.unappliedDb[T]
-        val ltt         = LightTypeTag(ref, fullDb, nameDb)
-        val serialized  = ltt.serialize()
-        val hashCodeRef = serialized.hash
-        val strRef      = serialized.ref
-        val strDBs      = serialized.databases
-        val version     = LightTypeTag.currentBinaryFormatVersion
-        val tagStr      = s"$hashCodeRef|$strRef|$strDBs|$version"
-        Expr(tagStr)
-    end tagImpl
+    private[Tag] object internal:
+
+        // Example Tag
+        //
+        // class Super
+        // class Param1 extends Super
+        // class Param2 extends Super
+        // class Test[-T, +U]
+        //
+        // Tag[Test[Param1, Param2]]
+        //
+        // Test;Super;java.lang.Object;scala.Matchable;scala.Any;[-Param1;Super;java.lang.Object;scala.Matchable;scala.Any;,+Param2;Super;java.lang.Object;scala.Matchable;scala.Any;]
```

Comment:
This format has some opportunity for optimization. We might be able to omit `java.lang.Object;scala.Matchable;scala.Any` and assume they're always present but I'm not sure about `Matcheable`.